src/main/java/easytests/api/v1/controllers/SubjectsController.java

```diff
@@ -93,4 +93,16 @@ private SubjectModelInterface getSubjectModel(Integer id) throws NotFoundExcepti
     /**
      * delete(subjectId)
      */
+
+    @DeleteMapping("/{subjectId}")
+    public void delete(@PathVariable Integer subjectId) throws NotFoundException, ForbiddenException {
+        final SubjectModelInterface subjectModel = this.getSubjectModel(subjectId);
```

Comment:
get subject model will not fetch it for delete, you need to provide Options to select object for delete.
Select for delete (All topics, questions and etc will be included and linked)
Than, you need call delete with that option.


So:
You need first to get subjects for auth, verify user have access and subjects exists.
Than select subject for delete.
Than delete subject with for delete options.